### ISSUE TITLE
Copose not setting up network with IPAM

### ISSUE THREAD

**Issue by rayjohnson** [docker/compose#5194]
I’m trying to create a network that will get an IP from a IPAM driver. The following docker command works great:

docker network create -d qnet --ipam-driver=qnet --ipam-opt=iface=eth1 qnet-dhcp-eth1

However, I’d like to have this work from my compose file. Here is what I have for the network section of the compose file:

```
networks:
  back:
    driver: customer-driver-1
  front:
    ipam:
      driver: qnet
      options:
        iface: eth1
```
When I try this, however, I get a weird error:
Creating network “mywordpress_front” with the default driver
ERROR: ipv4 pool is empty
DESKTOP-8O0TNFC

Do I have the syntax wrong in the compose file?

Or is it a bug?  Or worse - not supported?

Ray

**Comment by rayjohnson** [comment]
Ok - I did that - here is what I got:

```
docker-compose up -d 

time="2017-09-18T16:52:38.836417805-07:00" level=debug msg="Calling GET /v1.22/info" 
time="2017-09-18T16:52:38.849270495-07:00" level=debug msg="Calling GET /v1.22/networks/mywordpress_front" 
time="2017-09-18T16:52:38.849462303-07:00" level=error msg="Handler for GET /v1.22/networks/mywordpress_front returned error: network mywordpress_front not found" 
time="2017-09-18T16:52:38.857195911-07:00" level=debug msg="Calling POST /v1.22/networks/create" 
time="2017-09-18T16:52:38.857349931-07:00" level=debug msg="form data: {\"IPAM\":{\"Config\":[],\"Driver\":\"qnet\",\"Options\":{\"iface\":\"eth1\"}},\"Name\":\"mywordpress_front\"}" 
time="2017-09-18T16:52:38.857598739-07:00" level=debug msg="Allocating IPv4 pools for network mywordpress_front (c0c2e1287f2620671ce363e64ea7dd67624f412c523e4a4005d5b629eb0faa9c)" 
time="2017-09-18T16:52:40.501142587-07:00" level=debug msg="releasing IPv4 pools from network mywordpress_front (c0c2e1287f2620671ce363e64ea7dd67624f412c523e4a4005d5b629eb0faa9c)" 
time="2017-09-18T16:52:41.949146089-07:00" level=error msg="Handler for POST /v1.22/networks/create returned error: ipv4 pool is empty" 

docker network create -d qnet --ipam-driver=qnet --ipam-opt=iface=eth1 qnet-test

time="2017-09-18T16:53:58.950511258-07:00" level=debug msg="Calling GET /_ping" 
time="2017-09-18T16:53:58.953149832-07:00" level=debug msg="Calling POST /v1.23/networks/create" 
time="2017-09-18T16:53:58.953394716-07:00" level=debug msg="form data: {\"Attachable\":false,\"CheckDuplicate\":true,\"ConfigFrom\":null,\"ConfigOnly\":false,\"Driver\":\"qnet\",\"EnableIPv6\":false,\"IPAM\":{\"Config\":[],\"Driver\":\"qnet\",\"Options\":{\"iface\":\"eth1\"}},\"Ingress\":false,\"Internal\":false,\"Labels\":{},\"Name\":\"qnet-test\",\"Options\":{},\"Scope\":\"\"}" 
time="2017-09-18T16:53:58.953661560-07:00" level=debug msg="Allocating IPv4 pools for network qnet-test (c45a4a18c32c3ce97a55ac5868af89b0d61416c9e65706f3c541188d204bdeb8)" 
```

First thing I noticed is the version of the API being called is different.  Not sure why...

I hope this helps?  Nothing jumps out at me...

Ray

**Comment by rayjohnson** [comment]
In case you ask which versions of docker and compose I'm running:

```
DESKTOP-8O0TNFC:mywordpress rayj$ docker --version
Docker version 17.06.2-ce, build cec0b72
DESKTOP-8O0TNFC:mywordpress rayj$ docker-compose --version
docker-compose version 1.16.1, build 6d1ac21
```

**Comment by rayjohnson** [comment]
Ok.  I figured this out and got things working.  The main difference between the CLI and docker-compose is that the CLI is filling out default values.  The CLI set the network driver to the same as the IPAM driver.  Unexpected to me.  But in any case, that was exactly the thing I needed to do in my case.  I set the network driver to qnet and things started to work.

I do think it is a bug that docker-compose does not fill out the defaults.  This will otherwise leave the defaults to the plugin to decide.  Which in turn will almost ensure the defaults will be different across drivers.

I also noticed the CheckDuplicate=true in the CLI.  However, I thought I read that the default for that was supposed to be false.  (Maybe that changed and I was reading the wrong version of the docs.). Otherwise, that is perhaps a bug in the docs or the code.

Ray
